pkg/model/group.go

```diff
@@ -2,22 +2,26 @@ package model
 
 import (
 	"gorm.io/gorm"
+	"time"
 )
 
 const AdministratorGroupName = "administrators"
 
 // Group domain object defining a group
 // swagger:model
 type Group struct {
-	gorm.Model
-	Name                 string               `gorm:"unique;"`
+	Name                 string               `gorm:"primarykey; unique;"`
 	Hostname             string               `gorm:"unique;"`
 	Users                []User               `gorm:"many2many:user_groups;constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
 	ClusterConfiguration ClusterConfiguration `json:"clusterConfiguration"`
+
+	CreatedAt time.Time
+	UpdatedAt time.Time
+	DeletedAt gorm.DeletedAt `gorm:"index"`
```

Comment:
Because they were already there. Before we decided to make Name the primary key, gorm.Model was embedded and thereby adding the above columns.

I believe DeletedAt is indexed because gorm does a soft delete unless you specify otherwise. So whenever something is selected it's actually only returned if DeletedAt isn't populated... At least that would be my guess :-)

Reply:
interesting. For this PR this seems ok then. Do we even want soft deletion?

Reply:
Probably not